### PR TITLE
Add configuration of migration options and nesting on instances and profiles

### PR DESCRIPTION
**Pull request by edlerd**
## Done

- Added "migration" configuration options on instances and profiles
- Added nesting option in security section for instances and profiles

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - create a profile and check the new migration configuration options as well ass security>nesting work well
    - edit a profile and check the new migration configuration options as well ass security>nesting work well
    - create an instance and check the new migration configuration options as well ass security>nesting work well
    - edit an instance and check the new migration configuration options as well ass security>nesting work well
    - create a vm and check the new migration configuration options as well ass security>nesting work well
    - edit a vm and check the new migration configuration options as well ass security>nesting work well

## Screenshots

image

### CODE REVIEW

**Comment by mas-who** [comment]
QA looks good so far, couldn't break anything yet.

File: src/util/configInheritance.tsx [hunk 1]
```diff
@@ -196,6 +197,15 @@ const getInstanceProfileProjectDefaults = (
     }
   }
 
+  if (configKey === "migration.stateful") {
```
Comment (mas-who) [inline src/util/configInheritance.tsx:200]: shall we leave a comment here to say this config can be set at the server, profile and instance level with that order of preference?

File: src/pages/instances/forms/InstanceFormMenu.tsx [hunk 2]
```diff
@@ -104,6 +105,7 @@ const InstanceFormMenu: FC<Props> = ({
               <MenuItem label={RESOURCE_LIMITS} {...menuItemProps} />
               <MenuItem label={SECURITY_POLICIES} {...menuItemProps} />
               <MenuItem label={SNAPSHOTS} {...menuItemProps} />
+              <MenuItem label={MIGRATION} {...menuItemProps} />
```
Comment (mas-who) [inline src/pages/instances/forms/InstanceFormMenu.tsx:108]: should we hide this menu item if we are creating a container?

**Comment by mas-who** [comment]
@edlerd code looks good, just some small nit so feel free to skip.

**Review by mas-who — APPROVED** [review]

File: src/pages/instances/forms/InstanceFormMenu.tsx (hunk 2 above)
Comment (edlerd) [reply src/pages/instances/forms/InstanceFormMenu.tsx:108]: I was thinking about this, but then didn't want to make the menu options disappear while configuring.

Comment (mas-who) [reply src/pages/instances/forms/InstanceFormMenu.tsx:108]: yeah you are right, I think having the setting disabled makes more sense then :+1:

**Comment by mas-who** [comment]
LGTM, thanks for the nice comment :+1:

**Pull request merged**